tests/unit/adapter/test_job_service.py

```diff
@@ -105,3 +106,55 @@ def test_get_maintenance_status(requests_mock: RequestsMocker):
     )
     maintenance_status = job_service.get_maintenance_status()
     assert maintenance_status.paused is False
+
+
+def test_get_maintenance_status_error(requests_mock: RequestsMocker):
+    requests_mock.get(
+        f"{JOB_SERVICE_URL}/maintenance-status",
+        status_code=500,
+        text=ERROR_RESPONSE,
+    )
+    with pytest.raises(HttpResponseError) as e:
+        job_service.get_maintenance_status()
+    assert ERROR_RESPONSE in str(e)
+
+
+@pytest.mark.parametrize(
+    "is_paused,expected_result",
+    [
+        (
+            True,
+            {
+                "built_jobs": [],
+                "queued_manager_jobs": [],
+                "queued_worker_jobs": [],
+            },
+        ),
+        (
+            False,
+            {
+                "built_jobs": JOB_LIST,
+                "queued_manager_jobs": JOB_LIST,
+                "queued_worker_jobs": JOB_LIST,
+            },
+        ),
+    ],
+)
+def test_query_for_jobs(
+    is_paused, expected_result, requests_mock, monkeypatch
+):
+    # Mock the is_system_paused method
+    monkeypatch.setattr("job_executor.app.is_system_paused", lambda: is_paused)
+
+    def mock_get_jobs(job_status=None, operations=None):
```

Comment:
This would not be an accurate mocking of job-service 👍🏻 Job service still returns jobs from the /jobs endpoint even if it is paused 👍🏻

The goal of the query_for_jobs function should be:
* If job-service reports being paused, do not query for jobs
* If job-service reports not being paused, query for jobs